xarray/backends/api.py

```diff
@@ -570,6 +576,9 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
     parallel : bool, optional
         If True, the open and preprocess steps of this function will be
         performed in parallel using ``dask.delayed``. Default is False.
+    combine : {'auto', 'manual'}, optional
+        Whether ``xarray.auto_combine`` or ``xarray.manual_combine`` is used to
+        combine all the data. Default is 'auto'.
```

Comment:
This seems unnecessary. If a user supplies values for `concat_dim`, then it explicitly means manual. If the user does not supply `concat_dim`, then it means auto. The logic even precludes requesting auto, and supplying concat_dim and whether the inverse would work (requesting manual and not supplying concat_dim) isn't really clear.

Reply:
I think you make a good point here. It relates back to me saying 

>We didn't decide whether the ability to infer concatenation dimensions (i.e. when concat_dim='_CONCAT_DIM_DEFAULT') should be retained or not. 

At the moment then if `combine='manual'` and `concat_dim` is not supplied then it will try to infer the concatenation dimension for every "dimension" in the nested-list-input. If we require that `concat_dims` is supplied explicitly when `combine='manual'` then we are free to do what you suggest.